### PR TITLE
Setup code coverage

### PR DESCRIPTION
**Pull request by alin-turcu**
Setup code coverage #2

### CODE REVIEW

**Comment by stephanenicolas** [comment]
RTM. Pretty cool

File: build.gradle [hunk 1]
```diff
@@ -53,6 +54,45 @@ subprojects { project ->
...
+  }
+
+  /*
+   * Builds list of source dirs, class dirs, and executionData files
+   * when task is run, not at script evaluation time
+   */
+  doFirst {
+    subprojects.findAll { subproject ->
+      //we care only if the project is a library and is a rx2 implementation
+      !subproject.pluginManager.hasPlugin('com.android.application') && !subproject.name.endsWith("rx")
```
Comment (saguinav) [inline build.gradle]: why only rx2?

File: build.gradle [hunk 2]
```diff
@@ -33,6 +33,7 @@ buildscript {
 }
 
 allprojects {
+  apply plugin: 'jacoco'
```
Comment (saguinav) [inline build.gradle]: Should this only be applied for library projects?

File: build.gradle [hunk 3]
```diff
@@ -53,6 +54,45 @@ subprojects { project ->
...
+   */
+  doFirst {
+    subprojects.findAll { subproject ->
+      //we care only if the project is a library and is a rx2 implementation
+      !subproject.pluginManager.hasPlugin('com.android.application') && !subproject.name.endsWith("rx")
+    }.each { subproject ->
+      additionalSourceDirs files((Set<File>) subproject.sourceSets.main.allJava.srcDirs)
+      additionalClassDirs((FileCollection) subproject.sourceSets.main.output)
+      if (subproject.pluginManager.hasPlugin('jacoco')) {
+        executionData subproject.tasks.jacocoTestReport.executionData
```
Comment (saguinav) [inline build.gradle]: I'm wondering if we could place the `jacocoTestRepor` extension setup here.

```groovy
jacocoTestReport {
   dependsOn subproject.tasks.test
   reports {
      xml.enabled true
      html.enabled false
   }
}
rootProject.tasks.jacocoTestReport.dependsOn subproject.tasks.test
```

File: grox-commands-rx2/build.gradle [hunk 4]
```diff
@@ -23,3 +23,12 @@ license {
     java = 'SLASHSTAR_STYLE'
   }
 }
+
+jacocoTestReport {
```
Comment (saguinav) [inline grox-commands-rx2/build.gradle]: Check my comment above. We may centralize this part in the root build.gradle file.

File: build.gradle [hunk 5]
```diff
@@ -33,6 +33,7 @@ buildscript {
 }
 
 allprojects {
```
Comment (saguinav) [inline build.gradle]: This should be `subprojects`. We don't want to apply `jacoco` plugin to the root project.

File: build.gradle [hunk 6]
```diff
@@ -53,6 +54,44 @@ subprojects { project ->
   version = VERSION_NAME
 }
 
+apply plugin: 'jacoco'
```
Comment (saguinav) [inline build.gradle]: Is it needed to apply jacoco plugin to the root project?

File: config/jacoco_java.gradle [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  dependsOn tasks.test
+
+  reports {
+    html.enabled = true
+    xml.enabled = true
+    csv.enabled = false
+  }
+}
+
+rootProject.tasks.jacocoTestReport.dependsOn tasks.test
```
Comment (saguinav) [inline config/jacoco_java.gradle]: EOF

**Comment by saguinav** [comment]
2 minor comments. Feel free to address them or not 🚢

File: build.gradle (hunk 6 above)
Comment (alin-turcu) [reply build.gradle]: It doesn't know how to generate the jacoco task without this.

**Pull request merged**